numba/experimental/jitclass/overloads.py

```diff
@@ -0,0 +1,169 @@
+"""
+Overloads for ClassInstanceType for built-in functions that call dunder methods
+on an object.
+"""
+import operator
+import sys
+
+from numba.core.extending import overload
+from numba.core.types import ClassInstanceType
+
+
+def _get_args(nargs=1):
+    return list("xyzabcdefg")[:nargs]
+
+
+def _simple_template(*attrs, nargs=1):
+    args = _get_args(nargs=nargs)
+    arg0 = args[0]
+
+    template = f"""
+def handler({",".join(args)}):
+    if not isinstance({arg0}, ClassInstanceType):
+        return
+"""
+    for attr in attrs:
+        assert isinstance(attr, str)
+        template += f"""
+    if "__{attr}__" in {arg0}.jit_methods:
+        return lambda {",".join(args)}: {arg0}.__{attr}__({",".join(args[1:])})
+"""
+    return template
+
+
+def _register_overload(func, template, glbls=None):
+    """
+    Register overload handler for func calling class attribute attr.
+    """
+    namespace = dict(ClassInstanceType=ClassInstanceType)
+    if glbls:
+        namespace.update(glbls)
+
+    exec(template, namespace)
+    handler = namespace["handler"]
```

Comment:
Is this necessary, vs the simpler:
```suggestion
    dunder = f"__{attr}__"
    def handler(self, *args):
        if not isinstance(self, ClassInstanceType):
            return
        if dunder in jit.methods:
            return lambda self, *args: getattr(self, dunder)(*args)
```

Reply:
Unfortunately `getattr` is not currently supported, so that would not work.